Dockerfile

```diff
@@ -25,4 +25,12 @@ RUN    apt-get update                \
             sudo                     \
             zlib1g-dev
 
+# install more recent version of crypto++ because Debian derivatives package
+# libcrypto++ version 5.6.4 incorrectly so that the blake2 hash function can be
+# compiled but not invoked at link time
+RUN    curl "http://http.us.debian.org/debian/pool/main/libc/libcrypto++/libcrypto++8_8.2.0-2_amd64.deb"    > libcpp.deb    \
+    && curl "http://http.us.debian.org/debian/pool/main/libc/libcrypto++/libcrypto++-dev_8.2.0-2_amd64.deb" > libcppdev.deb \
+    && dpkg -i libcpp.deb libcppdev.deb                                                                                     \
+    && rm libcpp.deb libcppdev.deb
```

Comment:
I think it's better to build from source then. You should:

-   Add a submodule for the library in `deps/libcryptopp`
-   Add a command to the `Makefile` for building the target libraries appropriately.
-   Make sure that all targets (like the client build) depend on the build products correctly (and `.a` or `.o` files they need) in the `Makefile` and `CMakeLists.txt`.